### ISSUE TITLE
Add pip installation

### ISSUE THREAD

**Issue by pklimov** [bbopt/nomad#111]
I found an [unofficial](https://pypi.org/project/SQNomad/) `pip` installation path to an older and possibly modified version of NOMAD. Would be convenient to have an official `pip` installation path to the latest version.

**Comment by ctribes** [comment]
I will push this feature up in the list of TODO.

**Comment by jan-provaznik** [comment]
I have looked at the possible approaches to PyPi distribution. 
- Ship pre-compiled wheels for common platforms. To make the result as user friendly as possible, one would have to link the extension statically against nomadUtil, nomadEval and nomadAlgo libraries. I have actually explored the viability of this approach and it is possible (and fairly easy) to link the entire project statically. No more dependence on shared libraries. LGPL should be a non-issue in this case, the code of the interface is licensed under LGPL already.
- Have the user compile the interface. This is possible with slight modification of the current `setup.py` code. The user would have to set `NOMAD_HOME` to their NOMAD installation directory.
- Have the user compile the interface from some modified source, same as the unofficial version referenced here does it.

**Comment by ctribes** [comment]
Thanks for this analysis. I really prefer the first approach : ship pre-compiled wheels for common platforms. I agree about the LGPL licence.
 
I suppose that, to obtain the binaries, your approach relies on a modified setup_PyNomad.py script ? How to modify it to obtain static link with nomad libraries ?
You can create a pull request to add a modified setup_PyNomadStatic.py.

**Comment by jan-provaznik** [comment]
In my experiment I modified the relevant `CMakeLists.txt` files to build both static and shared version of the `nomad{Algo,Eval,Util}` libraries. The approach I took compiles the underlying objects only once, the compilation time does _not_ increase noticeably. I have then slightly lobotomized the `setup.py` script to do my bidding. 

A slight problem is I have not tested this on any other platform beside Linux. 
I'll put the necessary modifications together and make a pull request ASAP, the testing can be done once at least a proof of concept exists.
